File: src/test/java/Pages/AdminPage.java
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class AdminPage {

    WebDriver driver;


    By admin = By.id("menu_admin_viewAdminModule");
    By addUser = By.id("btnAdd");

    // Locators Add User

    By rolUser = By.id("systemUser_userType");
    By nameEmploy = By.id("systemUser_employeeName_empName");
    By userName = By.id("systemUser_userName");
    By status = By.id("systemUser_status");
    By tipoStatus = By.xpath("//*[@id='systemUser_status']/option[1]");
    By pass = By.id("systemUser_password");
    By confPass = By.id("systemUser_confirmPassword");
    By save = By.id("btnSave");
    By btnOk = By.id("dialogDeleteBtn");

    // Locator Delete User

    By userToDelete = By.id("searchSystemUser_userName");
    By delUserName = By.id("searchSystemUser_userName");
    By searchUser = By.id("searchBtn");
    By selectUser = By.id("ohrmList_chkSelectAll");
    By btnDelete = By.id("btnDelete");



    public AdminPage(WebDriver driver){

        this.driver = driver;
    }

    public void setAdmin(){
        driver.findElement(admin).click();
    }
    public void setAddUser(){
        driver.findElement(addUser).click();
    }
    public void setRolUser(){
        driver.findElement(rolUser).click();
    }
    public void setNameEmploy(){
        driver.findElement(nameEmploy).sendKeys("Fiona Grace");
    }
    public void setUserName(){
        driver.findElement(userName).sendKeys("FioGrace");
    }
    public void setStatus(){
        driver.findElement(status).click();
    }
    public void setTipoStatus(){
        driver.findElement(tipoStatus).click();
    }
    public void setPass(){
        driver.findElement(pass).sendKeys("Testeando15");
    }
    public void setConfPass(){
        driver.findElement(confPass).sendKeys("Testeando15");
    }
    public void setSave(){
        driver.findElement(save).click();
    }

    // Constructores Delete User

    public void setUserToDelete(){
        driver.findElement(userToDelete).click();
    }
    public void setDelUserName(){
        driver.findElement(delUserName).sendKeys("FioGrace");
    }
    public void setSearchUser(){
        driver.findElement(searchUser).click();
    }
    public void setSelectUser(){
        driver.findElement(selectUser).click();
    }
    public void setBtnDelete(){
        driver.findElement(btnDelete).click();
    }
    public void setBtnOk(){
        driver.findElement(btnOk).click();
    }
}

File: src/test/Kotlin/Pages/AdminPage.kt
package Pages

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy
import org.openqa.selenium.support.PageFactory

class AdminPage(driver: WebDriver) {

    init {
        PageFactory.initElements(driver, this)
    }

    @FindBy(id = "menu_admin_viewAdminModule")
    var admin: WebElement? = null

    @FindBy(id = "btnAdd")
    var addUser: WebElement? = null

    // Locators Add User
    @FindBy(id = "systemUser_userType")
    var rolUser: WebElement? = null
    @FindBy(id = "systemUser_employeeName_empName")
    var nameEmploy: WebElement? = null
    @FindBy(id = "systemUser_userName")
    var userName: WebElement? = null
    @FindBy(id = "systemUser_status")
    var status: WebElement? = null
    @FindBy(xpath = "//*[@id='systemUser_status']/option[1]")
    var tipoStatus: WebElement? = null
    @FindBy(id = "systemUser_password")
    var pass: WebElement? = null
    @FindBy(id = "systemUser_confirmPassword")
    var confPass: WebElement? = null
    @FindBy(id = "btnSave")
    var save: WebElement? = null
    @FindBy(id = "dialogDeleteBtn")
    var btnOk: WebElement? = null

    // Locator Delete User
    @FindBy(id = "searchSystemUser_userName")
    var userToDelete: WebElement? = null
    @FindBy(id ="searchSystemUser_userName")
    var delUserName: WebElement? = null
    @FindBy(id ="searchBtn")
    var searchUser: WebElement? = null
    @FindBy(id ="ohrmList_chkSelectAll")
    var selectUser: WebElement? = null
    @FindBy(id ="btnDelete")
    var btnDelete: WebElement? = null


    fun setAdmin() {
        admin?.click()
    }

    fun setAddUser() {
        addUser?.click()
    }

    fun setRolUser() {
        rolUser?.click()
    }

    fun setNameEmploy() {
        nameEmploy?.sendKeys("Lisa Andrews")
    }

    fun setUserName() {
        userName?.sendKeys("LisaAn")
    }

    fun setStatus() {
        status?.click()
    }

    fun setTipoStatus() {
        tipoStatus?.click()
    }

    fun setPass() {
        pass?.sendKeys("Selenium-.99")
    }

    fun setConfPass() {
        confPass?.sendKeys("Selenium-.99")
    }

    fun setSave() {
        save?.click()
    }

    // Constructores Delete User
    fun setUserToDelete() {
        userToDelete?.click()
    }

    fun setDelUserName() {
        delUserName?.sendKeys("LisaAn")
    }

    fun setSearchUser() {
        searchUser?.click()
    }

    fun setSelectUser() {
        selectUser?.click()
    }

    fun setBtnDelete() {
        btnDelete?.click()
    }

    fun setBtnOk() {
        btnOk?.click()
    }

}